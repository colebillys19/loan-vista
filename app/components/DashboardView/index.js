/**
 * DashboardView
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import DashboardBorrower from 'components/DashboardBorrower';
import DashboardCalls from 'components/DashboardCalls';
import DashboardDocuments from 'components/DashboardDocuments';
import DashboardLoan from 'components/DashboardLoan';
import DashboardPayments from 'components/DashboardPayments';
import TabContainer from 'components/_base-ui/TabContainer';

const DashboardView = (props) => {
  const {
    dashboardBorrowerData,
    dashboardListsData: { callsData, documentsData, paymentsData },
    dashboardLoanData,
    error,
    loading,
    loanNumber,
  } = props;
  const showComponents = loanNumber && !error && !loading;

  return (
    <TabContainer aria-labelledby="dashboard-tab" id="dashboard-view">
      <DashboardBorrower
        data={dashboardBorrowerData}
        renderLoading={!showComponents}
      />
      <DashboardLoan data={dashboardLoanData} renderLoading={!showComponents} />
      <DashboardCalls data={callsData} renderLoading={!showComponents} />
      <DashboardDocuments
        data={documentsData}
        renderLoading={!showComponents}
      />
      <DashboardPayments data={paymentsData} renderLoading={!showComponents} />
    </TabContainer>
  );
};

DashboardView.propTypes = {
  dashboardBorrowerData: T.array.isRequired,
  dashboardListsData: T.shape({
    callsData: T.array,
    documentsData: T.array,
    paymentsData: T.array,
  }).isRequired,
  dashboardLoanData: T.object.isRequired,
  error: T.oneOfType([T.bool, T.string]).isRequired,
  loading: T.bool.isRequired,
  loanNumber: T.string.isRequired,
};

export default DashboardView;
