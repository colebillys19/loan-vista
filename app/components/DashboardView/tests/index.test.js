import React from 'react';
import { render } from 'react-testing-library';

import DashboardView from '../index';

describe('<DashboardView />', () => {
  it('Expect not to log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<DashboardView />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<DashboardView />);
    expect(firstChild).toMatchSnapshot();
  });
});
