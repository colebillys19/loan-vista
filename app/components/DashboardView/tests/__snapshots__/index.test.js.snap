// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardView /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="dashboard-tab"
  class="TabContainer-sc-1us62xt-0 fqhtXT"
  id="dashboard-view"
>
  <section
    class="styledComponents__CardContainer-sc-1glo3a7-0 hVJZWB"
  >
    <div
      class="styledComponents__TopStripe-sc-1glo3a7-4 dlaTng"
      color="rgb(70, 99, 51)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1glo3a7-2 ihYfJV"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 4rem; width: 4rem;"
        viewBox="0 0 64 64"
      >
        <g>
          <path
            d="m3 55h44.08c0-8.208-8.561-13.58-16.774-14.986a11.1 11.1 0 1 0 -16.286-9.783 11.164 11.164 0 0 0 5.754 9.783c-8.258 1.414-16.774 6.806-16.774 14.986z"
            fill="rgb(255, 191, 0)"
          />
          <path
            d="m52.606 30.523a28.051 28.051 0 0 0 -3.974-1.6 13.158 13.158 0 0 0 -9.672-21.923 13.033 13.033 0 0 0 -12.655 10.165c-.417-.04-.838-.064-1.265-.064a13.158 13.158 0 0 0 -9.661 21.939c-7.505 2.414-14.379 7.918-14.379 15.96a2 2 0 0 0 2 2h44.08a2 2 0 0 0 2-2c0-8.042-6.874-13.547-14.379-15.96a13.191 13.191 0 0 0 -4.523-20.869 8.995 8.995 0 0 1 17.8 1.96 9.151 9.151 0 0 1 -4.717 8.031 2 2 0 0 0 .626 3.723 25.3 25.3 0 0 1 6.938 2.22c4.3 2.145 7.143 5.335 7.943 8.794h-10.483a19.852 19.852 0 0 1 2.783 4h9.932a2 2 0 0 0 2-2c0-5.755-3.883-11.128-10.394-14.376zm-18.546-.292a9.151 9.151 0 0 1 -4.717 8.031 2 2 0 0 0 .626 3.724c9.037 1.547 13.791 6.456 14.871 11.014h-39.6c1.08-4.558 5.834-9.467 14.871-11.014a2 2 0 0 0 .626-3.724 9.1 9.1 0 1 1 13.323-8.031z"
          />
        </g>
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1glo3a7-1 kipGlV"
      >
        Borrower Information
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1glo3a7-3 jEBAxK"
      />
    </div>
    <div
      class="styledComponents__ListsContainer-sc-1t59z9k-0 jjIreI"
    >
      <div
        class="styledComponents__StyledTwoColumnList-sc-1t59z9k-1 dFGcyd"
      >
        <h3
          class="styledComponents__ListHeading-oikso6-1 fKfFsr"
        >
          title
        </h3>
      </div>
    </div>
    <div
      class="GradientBorder-sc-1rv7gqq-0 yNJzF"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1glo3a7-0 hVJZWB"
  >
    <div
      class="styledComponents__TopStripe-sc-1glo3a7-4 kJMcOS"
      color="rgb(255, 191, 0)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1glo3a7-2 ihYfJV"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 4rem; width: 4rem;"
        viewBox="-34 0 504 504"
      >
        <path
          d="m16.035156 16h303.429688v370.855469h-303.429688zm0 0"
          fill="rgb(70, 99, 51)"
        />
        <path
          d="m436.640625 117.121094v370.878906c-.027344 8.824219-7.175781 15.972656-16 16h-303.441406c-8.835938.003906-16-7.164062-16-16v-117.121094h202.242187v-338.878906h-271.441406v338.878906h37.199219v32h-53.199219c-8.824219-.023437-15.9726562-7.171875-16-16v-370.878906c.0273438-8.824219 7.175781-15.9726562 16-16h303.441406c8.835938-.00390625 16 7.164062 16 16v370.878906c0 8.839844-7.164062 16.003906-16 16h-186.242187v69.121094h271.441406v-338.878906h-37.199219v-32h53.199219c8.824219.023437 15.972656 7.171875 16 16zm0 0"
        />
        <path
          d="m84.320312 101.144531h166.859376v32h-166.859376zm0 0"
        />
        <path
          d="m84.320312 185.429688h166.859376v32h-166.859376zm0 0"
        />
        <path
          d="m84.320312 269.714844h166.859376v32h-166.859376zm0 0"
        />
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1glo3a7-1 kipGlV"
      >
        Loan Information
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1glo3a7-3 jEBAxK"
      />
    </div>
    <div
      class="styledComponents__ListsContainer-sc-1881tcc-1 hUewNp"
    >
      <div
        class="styledComponents__StyledTwoColumnList-sc-1881tcc-3 kFbJqt"
      >
        <h3
          class="styledComponents__ListHeading-oikso6-1 fKfFsr"
        >
          title
        </h3>
      </div>
      <div
        class="styledComponents__StyledTwoColumnList-sc-1881tcc-3 kFbJqt"
      >
        <h3
          class="styledComponents__ListHeading-oikso6-1 fKfFsr"
        >
          title
        </h3>
      </div>
    </div>
    <div
      class="styledComponents__StopsListContainer-sc-1881tcc-2 gwwcxP"
    >
      <h3
        class="styledComponents__StopsListHeading-sc-1881tcc-0 bOejRM"
      >
        Stops
      </h3>
      <table
        class="styledComponents__StyledTable-qh5wjg-0 iUJMBY"
      >
        <tbody
          class="styledComponents__StyledTableBody-qh5wjg-1 efBlkm"
        />
      </table>
    </div>
    <div
      class="GradientBorder-sc-1rv7gqq-0 yNJzF"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1glo3a7-0 gwuMZd"
  >
    <div
      class="styledComponents__TopStripe-sc-1glo3a7-4 dlaTng"
      color="rgb(70, 99, 51)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1glo3a7-2 ihYfJV"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 4rem; width: 4rem;"
        viewBox="0 0 458 458"
      >
        <g>
          <path
            d="M427.456,308.232 C448.861448,337.575641 445.704184,378.138 420.016,403.816 C357.264,466.568 256.424,446.048 196.816,386.408 L71.568,261.2 C12,201.632 -8.688,100.832 54.16,38 C79.8436614,12.3171384 120.403867,9.16030592 149.752,30.56 C168.839267,44.5187871 178.927856,67.6766841 176.152,91.16 C172.68,119.832 151.496,128.32 142.752,137.064 C115.681918,164.194428 115.681918,208.117572 142.752,235.248 C216.744,304.304 224.728,329.72 262.248,334.912 C309.36,341.416 330.112,302.816 336.072,296.848 C361.384,271.504 405.664,278.544 427.456,308.232 Z"
            fill="rgb(255, 191, 0)"
          />
          <path
            d="M440.464,298.8 C413.2,261.688 357.472,252.88 324.8,285.576 C323.053273,287.452199 321.420992,289.431774 319.912,291.504 C311.424,302.656 295.576,323.504 264.456,319.12 C233.88,314.888 226.056,291.12 154.016,223.92 C133.177689,203.024766 133.177689,169.207234 154.016,148.312 C160.928,141.424 187.616,128.912 192.016,92.928 C195.426139,63.7221476 182.904699,34.9319994 159.216,17.512 C123.41122,-8.39954539 74.1107123,-4.57698697 42.728,26.544 C-27.864,97.144 -3.2,209.2 60.144,272.56 L113.456,325.872 L136.096,303.224 L82.784,249.92 C28.344,195.464 9.864,104.704 65.368,49.2 C85.534126,29.2089235 117.201479,26.7383488 140.224,43.36 C154.633787,53.9220174 162.268864,71.4032898 160.224,89.152 C157.64,110.48 141.888,115.192 131.368,125.704 C115.32544,141.831369 106.353196,163.67539 106.428226,186.42292 C106.503256,209.170451 115.619402,230.954809 131.768,246.976 C205.712,315.992 215.864,344.736 260.072,350.864 C309.744,357.688 335.656,323.712 345.4,310.864 C346.304,309.672 347,308.68 347.44,308.2 C366,289.64 398.896,296.248 414.64,317.752 C431.264252,340.776409 428.786506,372.44891 408.784,392.608 C352.176,449.208 260.528,427.64 208.064,375.192 L158.728,325.856 L136.088,348.496 L185.432,397.832 C251.336,463.752 362.768,483.928 431.424,415.248 C462.522081,383.871722 466.34712,334.599798 440.464,298.8 L440.464,298.8 Z"
            fill="rgb(17, 17, 17)"
          />
        </g>
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1glo3a7-1 kipGlV"
      >
        Calls
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1glo3a7-3 jEBAxK"
      />
    </div>
    <div
      class="styledComponents__Message-sc-29kat5-0 jGksjJ"
    >
      No data to display.
    </div>
    <table
      class="styledComponents__StyledTable-qh5wjg-0 iUJMBY"
    >
      <tbody
        class="styledComponents__StyledTableBody-qh5wjg-1 efBlkm"
      />
    </table>
    <div
      class="styledComponents__ViewMoreContainer-hcrd2u-0 cNcyjo"
    >
      <button
        class="MuiButtonBase-root styledComponents__StyledButton-sc-9o47wt-0 edMUeY"
        tabindex="0"
        type="button"
      >
        View More
      </button>
    </div>
    <div
      class="GradientBorder-sc-1rv7gqq-0 yNJzF"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1glo3a7-0 gwuMZd"
  >
    <div
      class="styledComponents__TopStripe-sc-1glo3a7-4 kJMcOS"
      color="rgb(255, 191, 0)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1glo3a7-2 ihYfJV"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 4rem; width: 4rem;"
        viewBox="0 0 64 64"
      >
        <g>
          <path
            d="m32 31.999-29-19.999v40h58v-40z"
            fill="rgb(70, 99, 51)"
          />
          <path
            d="m61 10.013v-.013h-58v.013a1.977 1.977 0 0 0 -2 1.987v40a2 2 0 0 0 2 2h58a2 2 0 0 0 2-2v-40a1.977 1.977 0 0 0 -2-1.987zm-6.423 3.987-22.577 15.569-22.577-15.569zm4.423 36h-54v-34.191l16.3 11.241-12.3 8.609v4.882l15.813-11.068 6.051 4.173a2.007 2.007 0 0 0 2.272 0l6.051-4.173 19.813 13.868zm0-11.541-16.3-11.409 16.3-11.241z"
            fill="rgb(17, 17, 17)"
          />
        </g>
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1glo3a7-1 kipGlV"
      >
        Documents
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1glo3a7-3 jEBAxK"
      />
    </div>
    <div
      class="styledComponents__Message-sc-29kat5-0 jGksjJ"
    >
      No data to display.
    </div>
    <table
      class="styledComponents__StyledTable-qh5wjg-0 iUJMBY"
    >
      <tbody
        class="styledComponents__StyledTableBody-qh5wjg-1 efBlkm"
      />
    </table>
    <div
      class="styledComponents__ViewMoreContainer-hcrd2u-0 cNcyjo"
    >
      <button
        class="MuiButtonBase-root styledComponents__StyledButton-sc-9o47wt-0 edMUeY"
        tabindex="0"
        type="button"
      >
        View More
      </button>
    </div>
    <div
      class="GradientBorder-sc-1rv7gqq-0 yNJzF"
    />
  </section>
  <section
    class="styledComponents__CardContainer-sc-1glo3a7-0 gwuMZd"
  >
    <div
      class="styledComponents__TopStripe-sc-1glo3a7-4 dlaTng"
      color="rgb(70, 99, 51)"
    />
    <div
      class="styledComponents__SummaryTitleContainer-sc-1glo3a7-2 ihYfJV"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="height: 4rem; width: 4rem;"
        viewBox="0 0 64 64"
      >
        <g>
          <path
            d="M53,13.5H11a8.031,8.031,0,0,1-8,8.062V42.438A8.031,8.031,0,0,1,11,50.5H49L61,38.407V21.562A8.031,8.031,0,0,1,53,13.5Z"
            fill="rgb(255, 191, 0)"
          />
          <path
            d="M34.007,42.357V40.116A5.1,5.1,0,0,0,33.043,30H30.972a1.107,1.107,0,1,1,0-2.214H36.15v-4H34.007V21.643h-4v2.241A5.1,5.1,0,0,0,30.972,34h2.071a1.107,1.107,0,0,1,0,2.214h-5.18v4h2.143v2.143Z"
            fill="rgb(17, 17, 17)"
          />
          <path
            d="M1.007,11.5v41H49.833L63,39.23V27.5H59v8.9H47V48.5H12.809a10.062,10.062,0,0,0-7.8-7.86V23.36a10.062,10.062,0,0,0,7.8-7.86H51.2A10.064,10.064,0,0,0,59,23.36h4V11.5ZM51,40.406h5.2L51,45.645ZM8.665,48.5H5.007V44.791A6.055,6.055,0,0,1,8.665,48.5ZM5.007,19.209V15.5H8.665A6.055,6.055,0,0,1,5.007,19.209Zm53.993,0A6.053,6.053,0,0,1,55.343,15.5H59Z"
            fill="rgb(17, 17, 17)"
          />
        </g>
      </svg>
      <h2
        class="styledComponents__StyledH2-sc-1glo3a7-1 kipGlV"
      >
        Payments
      </h2>
      <div
        class="styledComponents__ThinStripe-sc-1glo3a7-3 jEBAxK"
      />
    </div>
    <div
      class="styledComponents__Message-sc-29kat5-0 jGksjJ"
    >
      No data to display.
    </div>
    <table
      class="styledComponents__StyledTable-qh5wjg-0 iUJMBY"
    >
      <tbody
        class="styledComponents__StyledTableBody-qh5wjg-1 efBlkm"
      />
    </table>
    <div
      class="styledComponents__ViewMoreContainer-hcrd2u-0 cNcyjo"
    >
      <button
        class="MuiButtonBase-root styledComponents__StyledButton-sc-9o47wt-0 edMUeY"
        tabindex="0"
        type="button"
      >
        View More
      </button>
    </div>
    <div
      class="GradientBorder-sc-1rv7gqq-0 yNJzF"
    />
  </section>
</section>
`;
